tests/helper.py

```diff
@@ -63,3 +66,66 @@ def create_in_memory_hdf5(filename="f1"):
     """
 
     return h5py.File(filename, driver="core", mode="a", backing_store=False)
+
+def create_single_particle_dataset(filename: str, output_name: str):
+    """
+    Create an hdf5 snapshot with two particles at an identical location
+
+    Parameters
+    ----------
+    filename: str
+        name of file from which to copy metadata
+    output_name: str
+        name of single particle snapshot
+    """
+    # Create a dummy mask in order to write metadata
+    data_mask = mask(filename)
+    boxsize = data_mask.metadata.boxsize
+    region = [[0, b] for b in boxsize]
+    data_mask.constrain_spatial(region)
+
+    # Write the metadata
+    infile = h5py.File(filename, "r")
+    outfile = h5py.File(output_name, "w")
+    list_of_links, _ = find_links(infile)
+    write_metadata(infile, outfile, list_of_links, data_mask)
+
+    # Write a single particle
+    particle_coords = cosmo_array([[1,1,1], [1,1,1]], data_mask.metadata.units.length)
+    particle_masses = cosmo_array([1, 1], data_mask.metadata.units.mass)
+    mean_h = mean(infile["/PartType0/SmoothingLengths"])
+    particle_h = cosmo_array([mean_h, mean_h], data_mask.metadata.units.length)
+    particle_ids = [1, 2]
+
+    coords = outfile.create_dataset("/PartType0/Coordinates", data=particle_coords)
+    for name, value in infile["/PartType0/Coordinates"].attrs.items():
+        coords.attrs.create(name, value)
+
+    masses = outfile.create_dataset("/PartType0/Masses", data=particle_masses)
+    for name, value in infile["/PartType0/Masses"].attrs.items():
+        masses.attrs.create(name, value)
+
+    h = outfile.create_dataset("/PartType0/SmoothingLengths", data=particle_h)
+    for name, value in infile["/PartType0/SmoothingLengths"].attrs.items():
+        h.attrs.create(name, value)
+
+    ids = outfile.create_dataset("/PartType0/ParticleIDs", data=particle_ids)
+    for name, value in infile["/PartType0/ParticleIDs"].attrs.items():
+        ids.attrs.create(name, value)
+
+    # Get rid of all traces of DM
+    del outfile["/Cells/Counts/PartType1"]
+    del outfile["/Cells/Offsets/PartType1"]
+    #nparts_total = outfile["/Header"].attrs.get("NumPart_Total")
+    #nparts_this_file = outfile["/Header"].attrs.get("NumPart_ThisFile")
+    nparts_total = [2, 0, 0, 0, 0, 0]
+    nparts_this_file = [2, 0, 0, 0, 0, 0]
+    outfile["/Header"].attrs.__setitem__("NumPart_Total", nparts_total)
```

Comment:
Why this and not just outfile["/Header"].attrs["NumPart_Total"] = nparts_total? Does that not work?

Reply:
Didn't know I could do that. Changed now.